src/App.tsx

```diff
@@ -7,38 +7,115 @@ import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { getUsers } from './api/users';
 import { Loader } from './components/Loader';
+import { getPosts } from './api/posts';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isProcessing, setIsProcessing] = useState(false);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const loadUsers = async () => {
+    try {
+      setIsLoadingError(false);
+
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch (error) {
+      setIsProcessing(false);
+      setIsLoadingError(true);
+    }
+  };
+
+  const getPostsList = async (user: User) => {
+    try {
+      setSelectedUser(user);
+      setIsProcessing(true);
+      setIsLoadingError(false);
+      setPosts(null);
+
+      const postsFromServer = await getPosts(user.id);
+
+      setPosts(postsFromServer);
+      setIsProcessing(false);
+    } catch (error) {
+      setIsProcessing(false);
+      setIsLoadingError(true);
+    }
+  };
+
+  const handlePostInfo = (postId: number) => {
+    if (posts) {
+      setSelectedPost(posts.find(post => post.id === postId) || null);
+    }
+
+    if (postId === 0) {
+      setIsSidebarOpen(false);
+    } else {
+      setIsSidebarOpen(true);
+    }
```

Comment:
You can use ternary operator here as follow:
`setState(statement ? true : false)`

Reply:
fixed